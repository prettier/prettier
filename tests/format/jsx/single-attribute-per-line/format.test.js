runFormatTest(import.meta, ["flow", "babel", "typescript"]);
runFormatTest(import.meta, ["flow", "babel", "typescript"], {
  singleAttributePerLine: true,
});
