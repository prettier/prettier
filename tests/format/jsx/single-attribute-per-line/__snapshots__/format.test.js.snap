// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single-attribute-per-line.js - {"singleAttributePerLine":true} format 1`] = `
====================================options=====================================
parsers: ["flow", "babel", "typescript"]
printWidth: 80
singleAttributePerLine: true
                                                                                | printWidth
=====================================input======================================
import React from "react";

const Component = () => (
  <div>
    <div data-a="1">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </div>

    <div data-a="1" data-b="2" data-c="3">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </div>

    <div data-a="Lorem ipsum dolor sit amet" data-b="Lorem ipsum dolor sit amet" data-c="Lorem ipsum dolor sit amet">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </div>

    <div data-long-attribute-a="1" data-long-attribute-b="2" data-long-attribute-c="3">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </div>

    <img src="/images/foo.png" />

    <img src="/images/foo.png" alt="bar" />

    <img src="/images/foo.png" alt="Lorem ipsum dolor sit amet, consectetur adipiscing elit." />
  </div>
);

=====================================output=====================================
import React from "react";

const Component = () => (
  <div>
    <div data-a="1">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </div>

    <div
      data-a="1"
      data-b="2"
      data-c="3"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </div>

    <div
      data-a="Lorem ipsum dolor sit amet"
      data-b="Lorem ipsum dolor sit amet"
      data-c="Lorem ipsum dolor sit amet"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </div>

    <div
      data-long-attribute-a="1"
      data-long-attribute-b="2"
      data-long-attribute-c="3"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </div>

    <img src="/images/foo.png" />

    <img
      src="/images/foo.png"
      alt="bar"
    />

    <img
      src="/images/foo.png"
      alt="Lorem ipsum dolor sit amet, consectetur adipiscing elit."
    />
  </div>
);

================================================================================
`;

exports[`single-attribute-per-line.js format 1`] = `
====================================options=====================================
parsers: ["flow", "babel", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
import React from "react";

const Component = () => (
  <div>
    <div data-a="1">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </div>

    <div data-a="1" data-b="2" data-c="3">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </div>

    <div data-a="Lorem ipsum dolor sit amet" data-b="Lorem ipsum dolor sit amet" data-c="Lorem ipsum dolor sit amet">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </div>

    <div data-long-attribute-a="1" data-long-attribute-b="2" data-long-attribute-c="3">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </div>

    <img src="/images/foo.png" />

    <img src="/images/foo.png" alt="bar" />

    <img src="/images/foo.png" alt="Lorem ipsum dolor sit amet, consectetur adipiscing elit." />
  </div>
);

=====================================output=====================================
import React from "react";

const Component = () => (
  <div>
    <div data-a="1">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </div>

    <div data-a="1" data-b="2" data-c="3">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </div>

    <div
      data-a="Lorem ipsum dolor sit amet"
      data-b="Lorem ipsum dolor sit amet"
      data-c="Lorem ipsum dolor sit amet"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </div>

    <div
      data-long-attribute-a="1"
      data-long-attribute-b="2"
      data-long-attribute-c="3"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </div>

    <img src="/images/foo.png" />

    <img src="/images/foo.png" alt="bar" />

    <img
      src="/images/foo.png"
      alt="Lorem ipsum dolor sit amet, consectetur adipiscing elit."
    />
  </div>
);

================================================================================
`;
